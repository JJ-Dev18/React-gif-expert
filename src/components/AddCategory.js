import React, { useState } from 'react'
import Proptypes from 'prop-types'


export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
   

      const handleInputChange = (e) => {
         console.log(e.target.value);
         setInputValue(e.target.value);

         console.log('HandleInputChange llamado')
         
     }

     const handleSubmit = (e) =>{
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }
     }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Add Category</h2>
           
             <input 
             type="text"
             value={inputValue}
             onChange={ handleInputChange}
             ></input>
             
        </form>
    )

   
}

AddCategory.propTypes = {

    setCategories : Proptypes.func.isRequired

}



