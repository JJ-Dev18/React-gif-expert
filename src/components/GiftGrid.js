import React  from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

// import { getGifs } from '../Helpers/getGifs';


export const GiftGrid = ({category}) => {

    // const [images, setImages] = useState([])
    const {data: images ,loading} = useFetchGifs(category);
  
    // useEffect(() => {
    //    getGifs(category).then( imgs => setImages(imgs))
    // },[category])
   

    return (
       <>
         <h3 className="animate__animated animate__rubberBand">{category}</h3>
         {loading && <p className="animate__animated animate__flash"> Loading ...</p>}

        
         <div className="card-grid">  
               {
                  images.map(img =>(
                    <GifGridItem 
                    key={img.id}
                    {...img}/>
                  )) 
               }     
        </div>
       </>
        
    )
}

GiftGrid.propTypes = {

   category : PropTypes.string.isRequired
}