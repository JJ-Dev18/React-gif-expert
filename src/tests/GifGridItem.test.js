import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import {GifGridItem} from '../components/GifGridItem'

describe('Pruebas en grifGridItem', () => {
    
    const id = '1asdf';
    const title = 'El mejor';
    const url = "asfraraasdaf";
    const wrapper = shallow( 
    <GifGridItem 
        id={id} 
        title={title} 
        url={url}
        />)
    test('Debe mostrar <gifgriditem /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title ', () => {
        
          const p = wrapper.find('p')
          expect(p.text().trim()).toBe(title)
    });
    
    test('Debe de tener la imagen igual al url de los props  ', () => {
        
        const img = wrapper.find('img')
       expect(img.props().src).toBe(url)
        
    });

    test('debe tener animate__fadein', () => {

        const div = wrapper.find('div');

        console.log(div.hasClass('animate__rubberBand'))
        expect(div.hasClass('animate__rubberBand')).toBe(true)
        
    })
    
    
    
});

