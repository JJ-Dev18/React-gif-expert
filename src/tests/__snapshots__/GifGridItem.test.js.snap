// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en grifGridItem Debe mostrar <gifgriditem /> correctamente 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="El mejor"
    src="asfraraasdaf"
  />
  <p>
    El mejor
  </p>
</div>
`;
